### PR TITLE
feat(failinject): add failinject feature

### PR DESCRIPTION
**Pull request by marlonfan**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pixiu/pkg/filter/failinject/filter.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+func (factory *FilterFactory) Apply() error {
+	return nil
+}
+
+func (factory *FilterFactory) Config() interface{} {
+	return factory.cfg
+}
+
+func (factory *FilterFactory) PrepareFilterChain(ctx *contextHttp.HttpContext, chain filter.FilterChain) error {
+	fmt.Println(factory.cfg)
```
Comment (mark4z) [inline pixiu/pkg/filter/failinject/filter.go]: remove debug info

File: pixiu/pkg/filter/failinject/filter.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	if uriCfg.Type == TypeDelay {
+		time.Sleep(uriCfg.Delay)
+		ctx.SendLocalReply(uriCfg.StatusCode, []byte(uriCfg.Body))
+		return filter.Stop
+	}
+
+	return filter.Continue
+}
+
+// 当前请求是否命中判断
```
Comment (mark4z) [inline pixiu/pkg/filter/failinject/filter.go]: 注解去掉中文或者写双语吧

**Comment by AlexStocks** [comment]
待 梦超 测试

File: pixiu/pkg/filter/failinject/filter.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package failinject
+
+import (
+	"math/rand"
```
Comment (AlexStocks) [inline pixiu/pkg/filter/failinject/filter.go:21]: 格式化下

**Review by mark4z — APPROVED** [review]

**Pull request merged**